### PR TITLE
refactor: granularity in the installation

### PR DESCRIPTION
**Pull request by sebastianpfischer**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -41,23 +41,55 @@ classifiers = [
...
+    "pyserial",
+]
+all = [
+    "pylink-square",
+    "pykiso-python-uds",
+    "python-can",
+    "pyserial",
+    "PyVISA",
+    "PyVISA-py",
+]
```
Comment (kakooloukia) [inline pyproject.toml:89]: Is it possible to re-use variables with poetry ?
eg:
all = can + debugger + instrument + serial

what is the difference between 'all' and 'plugins' ?

Comment (sebastianpfischer) [reply pyproject.toml:89]: Hey!
So @Pog3k answered your question :)
For the last point. Next step would be to separate the different tools from the core part. Which mean that there will be a difference between `plugins` and `all`.

File: src/pykiso/lib/connectors/cc_serial.py [hunk 2]
```diff
@@ -23,7 +23,11 @@
 from enum import Enum, IntEnum
 from typing import ByteString, Dict, Optional, Union
 
-import serial
+try:
+    import serial
+except ImportError("Dependencies missing, please install 'pip install pykiso[serial]'"):
+    raise
```
[the remark this answers is not in the document]
Comment (sebastianpfischer) [reply src/pykiso/lib/connectors/cc_serial.py:29]: Fair ^^, I will apply as soon as I have some time

**Review by BKaDamien — APPROVED** [review]

**Review by kakooloukia — APPROVED** [review]

File: src/pykiso/lib/connectors/cc_pcan_can.py [hunk 3]
```diff
@@ -24,9 +24,13 @@
 from pathlib import Path
 from typing import Dict, Optional, Union
 
-import can
-import can.bus
-import can.interfaces.pcan.basic as PCANBasic
+try:
```
Comment (sebclrsn) [inline src/pykiso/lib/connectors/cc_pcan_can.py:27]:
```
mock_python_can = MagicMock(side_effect=ImportError)
with patch.dict('sys.modules', can=mock_python_can):
    import cc_pcan_can
```

Comment (sebastianpfischer) [reply src/pykiso/lib/connectors/cc_pcan_can.py:27]: I have a pretty dangerous hack!!!!

```python
def test_import(mocker):
    with pytest.raises(ImportError):
        sys.modules['can'] = None
        importlib.reload(cc_pcan_can)
    sys.modules['can'] = can
    importlib.reload(cc_pcan_can)
```

**Comment by sebastianpfischer** [comment]
The coverage is not reflecting the reality anymore, the improvement can be found here: https://app.codecov.io/gh/eclipse/kiso-testing/commit/58f0e4808752e6d05e39c3a236c8482720820074

**Comment by sebastianpfischer** [comment]
Somehow there is a mismatch between the black versions used in Jenkins and the one we use... I could commit because it passed black formatting ^^ but Jenkins complains it does not.

**Comment by sebastianpfischer** [comment]
Coverage of `cc_usb.py` should be done with #255

**Pull request merged**